src/Model/Profile.php

```diff
@@ -166,7 +166,16 @@ public static function load(App $a, $nickname, array $profiledata = [], $show_co
 			}
 		}
 
-		$profile = !empty($user['uid']) ? User::getOwnerDataById($user['uid'], false) : [];
+		if (empty($user['uid'])) {
+			$profile = [];
+		} else {
+			$profile = array_merge(
+				$user,
+				Profile::getByUID($user['uid']),
+				Contact::getById(Contact::getIdForURL(Strings::normaliseLink(DI::baseurl() . '/profile/' . $nickname), local_user()))
+			);
```

Comment:
These lines are causing all _my_ contacts to show up on the profile pages of people I'm friends with. Fixing now.

Reply:
Seems to be working now. Not great that we're smashing all these objects into one object, IMO, but disentangling them at this point seems like a much bigger undertaking than just doing it this way.